Ds3/Helpers/Streams/DisposableStream.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ * ******************************************************************************
+ *   Copyright 2014 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+using System;
+using System.IO;
+
+namespace Ds3.Helpers.Streams
+{
+    internal class DisposableStream : Stream, IDisposable
```

Comment:
Since this is specific to FileStreams we may want to call it DisposableFileStream instead